Sources/libhostmgr/BuildkiteScriptBuilder.swift

```diff
@@ -0,0 +1,166 @@
+    public mutating func addDependency(atPath path: String) {
+        self.dependencies.append(path)
+    }
+
+    /// Add an environment variable pair to the build script.
+    ///
+    /// If there's an existing environment variable with the same name, it will be overwritten.
+    public mutating func addEnvironmentVariable(named key: String, value: String) {
+        self.environmentVariables[key] = Value(wrapping: value)
+    }
+
+    /// Copy environment variables from the existing environment into the build script based on their prefix.
+    public mutating func copyEnvironmentVariables(
+        prefixedBy prefix: String,
+        from environment: [String: String] = ProcessInfo.processInfo.environment
+    ) {
+        environment
+            .filter { $0.key.starts(with: prefix) }
+            .forEach { key, value in
+                environmentVariables[key] = Value(wrapping: value)
+            }
+    }
+
+    /// Add a line to the build script.
+    ///
+    /// This typically takes the form of a single command (like `cp foo bar`).
+    public mutating func addCommand(_ command: String, _ arguments: String...) {
+        self.commands.append(Command(command: command, arguments: arguments))
+    }
+
+    /// Compile the build script into a single string
+    public func build() -> String {
+        return [
+            dependencies
+                .map(convertDependencyToSource)
+                .joined(separator: "\n"),
+            environmentVariables
+                .sorted { $0.0 < $1.0 }
+                .filter { !$0.value.rawValue.isEmpty }
+                .map(convertEnvironmentVariableToExport)
+                .joined(separator: "\n"),
+            commands
+                .map(escapeCommand)
+                .joined(separator: "\n")
+        ].joined(separator: "\n")
+    }
+
+    /// Helper that takes a path like `~/.bashrc` and make it into a bash `source` command.
+    ///
+    /// Escapes spaces in paths automatically.
+    ///
+    /// Example:
+    ///
+    /// ```
+    /// # Given a path of `~/.bashrc`:
+    /// source ~/.bashrc
+    /// ```
+    func convertDependencyToSource(_ path: String) -> String {
+        "source \(path.escapingSpaces)".trimmingWhitespace
+    }
+
+    /// Helper that takes an environment variable key/value pair to an `export` statement.
+    ///
+    /// Automatically quote-wraps the value and escapes quotes as needed.
+    ///
+    /// Example:
+    ///
+    /// ```
+    /// # Given `foo:bar`
+    /// export foo="bar"
+    /// ```
+    func convertEnvironmentVariableToExport(_ pair: (String, Value)) -> String {
+        return "export \(pair.0)=\"\(pair.1.escapedRepresentation)\"".trimmingWhitespace
+    }
+
+    /// Helper that wraps command escape logic for shorthand use in a `map` statement.
+    func escapeCommand(_ command: Command) -> String {
+        command.escapedText.trimmingWhitespace
```

Comment:
This `trimmingWhitespace` is redundant—already trimmed in `escapedText`.

Reply:
Addressed in 84ef54fddb47619238e8e997b4a03a5b295ce6c0